src/java.base/share/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -614,6 +616,12 @@ protected void connect(SocketAddress remote, int millis) throws IOException {
                 throw SocketExceptions.of(ioe, isa);
             }
         }
+        if (SocketConnectEvent.enabled()) {
+            long duration = SocketConnectEvent.timestamp() - start;
+            if (SocketConnectEvent.shouldCommit(duration)) {
+                SocketConnectEvent.commit(start, duration, isa.getHostString(), address.getHostAddress(), port, connected);
+            }
+        }
```

Comment:
Would it be possible to update the JBS or PR description to indicate if the intent is to record an event when the connection cannot be established? I'm asking the change will only record an event when a connection is successfully established ("connected" is always true here).

Reply:
Capturing all calls even if they threw an exception does seem pretty useful.  I'll update the JBS